### PR TITLE
fix: can't run multiple services on k8s

### PR DESCRIPTION
**Pull request by eliasscosta**
Fix Issue: https://github.com/woodpecker-ci/woodpecker/issues/3288

The way the pod service starts up makes it impossible to run two or more pipelines at the same time when we have a service section. 

The idea is to set the name of the service in the same way we did for the pod name. 

Pipeline: 

```yaml

services:
  mydb:
    image: mysql
    environment:
      - MYSQL_DATABASE=test
      - MYSQL_ROOT_PASSWORD=example
    ports:
      - 3306/tcp
steps:
  get-version:
    image: ubuntu
    commands:
      - ( apt update && apt dist-upgrade -y && apt install -y mysql-client 2>&1 )> /dev/null
      - sleep 30s # need to wait for mysql-server init
      - echo 'SHOW VARIABLES LIKE "version"' | mysql -uroot -hmydb test -pexample
```

Running more than one pipeline result:

image

### CODE REVIEW

File: pipeline/backend/kubernetes/service_test.go [hunk 1]
```diff
@@ -24,24 +24,24 @@ import (
...
-	assert.Equal(t, "awesome-service", name)
+	assert.Equal(t, "awesome-service-01he8bebctabr3kgk0qj36d2me-0", name)
 }
 
 func TestService(t *testing.T) {
 	expected := `
 	{
 	  "metadata": {
-	    "name": "bar",
+	    "name": "bar-01he8bebctabr3kgk0qj36d2me-0",
```
Comment (6543) [inline pipeline/backend/kubernetes/service_test.go:44]: the equestion is, how is the internal network "hostname" set ... as services should be reachable by name "bar" ...

File: pipeline/backend/kubernetes/service_test.go [hunk 2]
```diff
@@ -24,24 +24,24 @@ import (
...
 	    "namespace": "foo",
 	    "creationTimestamp": null
 	  },
```
Comment (6543) [inline pipeline/backend/kubernetes/service_test.go:47]: i guess we should add "hostnames" key to spec ...!

Comment (6543) [reply pipeline/backend/kubernetes/service_test.go:47]: https://kubernetes.io/docs/tasks/network/customize-hosts-file-for-pods/#adding-additional-entries-with-hostaliases

Comment (eliasscosta) [reply pipeline/backend/kubernetes/service_test.go:47]: The hostname remains the same, just the `step.Name` to use in the pipeline. 

Name with UUID is just for not conflict with others k8s resources

File: pipeline/backend/kubernetes/service_test.go (hunk 1 above)
Comment (eliasscosta) [reply pipeline/backend/kubernetes/service_test.go:44]: It's if the service step has the name bar, it's added in the hostAlias. https://github.com/woodpecker-ci/woodpecker/blob/main/pipeline/backend/kubernetes/kubernetes.go#L194

File: pipeline/backend/kubernetes/service_test.go (hunk 2 above)
Comment (6543) [reply pipeline/backend/kubernetes/service_test.go:47]: yes if I'm correct the hostname was never set as it should be ...

.. I have to setup an k3s again to test services with cube ...

Comment (6543) [reply pipeline/backend/kubernetes/service_test.go:47]: the question is if they do not work, if you could also add the fix in here or create a new pull

if it works now just fine already - I'm happy :)

Comment (eliasscosta) [reply pipeline/backend/kubernetes/service_test.go:47]: In my tests, it already works as expected. 

Here I enter two different pod steps running they set `mydb` for each pipeline. 

image

Comment (zc-devs) [reply pipeline/backend/kubernetes/service_test.go:47]: > if I'm correct the hostname was never set as it should be

It worked before changes mentioned in #3178. Service name was unique, but hostalias (/etc/hosts) contained just name and IP, like `10.10.10.10 postgres`.

File: pipeline/backend/kubernetes/pod_test.go [hunk 3]
```diff
@@ -51,7 +51,7 @@ func TestStepToPodName(t *testing.T) {
 	assert.EqualValues(t, "wp-01he8bebctabr3kg", name)
 	name, err = stepToPodName(&types.Step{UUID: "01he8bebctabr3kg", Name: "postgres", Type: types.StepTypeService})
 	assert.NoError(t, err)
-	assert.EqualValues(t, "postgres", name)
+	assert.EqualValues(t, "postgres-01he8bebctabr3kg", name)
```
Comment (zc-devs) [inline pipeline/backend/kubernetes/pod_test.go:54]: I would make it suffix as it was before. Something like `wp-01he8bebctabr3kg-postgres` / `wp-01he8bebctabr3kg-svc-postgres` / `wp-01he8bebctabr3kg-service-postgres`.

File: pipeline/backend/kubernetes/service_test.go [hunk 4]
```diff
@@ -24,24 +24,24 @@ import (
 )
 
 func TestServiceName(t *testing.T) {
-	name, err := serviceName(&types.Step{Name: "database"})
+	name, err := serviceName(&types.Step{Name: "database", UUID: "01he8bebctabr3kgk0qj36d2me-0"})
 	assert.NoError(t, err)
-	assert.Equal(t, "database", name)
+	assert.Equal(t, "database-01he8bebctabr3kgk0qj36d2me-0", name)
```
Comment (zc-devs) [inline pipeline/backend/kubernetes/service_test.go:29]: The same: I would make it suffix as it was before.

**Review by zc-devs — APPROVED** [review]

**Review by 6543 — APPROVED** [review]

**Pull request merged**